### PR TITLE
feat: adding photo type and more metadata fields from dvr recordings

### PR DESCRIPTION
**Pull request by russorat**
found this while poking around with photos:

```
{
  "Account": {
    "id": 1, 
    "thumb": "https://plex.tv/users/xxxxxxxxxxx/avatar?c=xxxxxxxxx", 
    "title": "xxxxxxxx"
  }, 
  "Metadata": {
    "addedAt": 1614209252, 
    "createdAtAccuracy": "local", 
    "createdAtTZOffset": "-28800", 
    "guid": "local://xxxx", 
    "index": 1, 
    "key": "/library/metadata/8369", 
    "librarySectionID": 4, 
    "librarySectionKey": "/library/sections/4", 
    "librarySectionTitle": "Photos", 
    "librarySectionType": "photo", 
    "originallyAvailableAt": "2021-02-24", 
    "ratingKey": "xxxx", 
    "summary": "", 
    "thumb": "/library/metadata/xxxx/thumb/xxxxxxxxxx", 
    "title": "Screen Shot 2021-02-24 at 3.26.14 PM", 
    "type": "photo", 
    "updatedAt": 1614209253, 
    "year": 2021
  }, 
  "Player": {
    "local": true, 
    "publicAddress": "xx.xx.xx.xx", 
    "title": "Chrome", 
    "uuid": "xxxxxxx"
  }, 
  "Server": {
    "title": "xxxxxxx", 
    "uuid": "xxxxxxx"
  }, 
  "event": "media.play", 
  "owner": true, 
  "user": true
}
```

### CODE REVIEW

**Comment by russorat** [comment]
some more from a dvr recording:

```
{
  "Account": {
    "id": 1, 
    "thumb": "https://plex.tv/users/xxxxxxxxxx/avatar?c=xxxxxxxxxx", 
    "title": "xxxxxxxx"
  }, 
  "Metadata": {
    "Channel": [
      {
        "count": 2, 
        "filter": "channel=33160", 
        "id": 33160, 
        "tag": "2.3 KTVUDT3 (Movies!)"
      }, 
      {
        "filter": "channel=33315", 
        "id": 33315, 
        "tag": "66.4 KFSFDT4 (Grit)"
      }
    ], 
    "Field": [
      {
        "locked": true, 
        "name": "thumb"
      }, 
      {
        "locked": true, 
        "name": "art"
      }
    ], 
    "Genre": [
      {
        "count": 19, 
        "filter": "genre=287", 
        "id": 287, 
        "tag": "Western"
      }
    ], 
    "addedAt": 1614242259, 
    "art": "/library/metadata/8372/art/1614242261", 
    "contentRating": "TVPG", 
    "duration": 7200000, 
    "guid": "plex://movie/xxxxxxxxx", 
    "key": "/library/metadata/8372", 
    "lastViewedAt": 1614273229, 
    "librarySectionID": 2, 
    "librarySectionKey": "/library/sections/2", 
    "librarySectionTitle": "Movies", 
    "librarySectionType": "movie", 
    "oneShot": "1", 
    "originallyAvailableAt": "1957-01-01", 
    "ratingKey": "8372", 
    "summary": "Dan Evans (Van Heflin), a drought-plagued Arizona rancher, volunteers to take captured stagecoach robber and murderer Ben Wade (Glenn Ford) from Bisbee to Contention City, where the criminal will be put aboard the 3:10 train to Yuma for his trial. Accompanied only by the town drunk, Alex Potter (Henry Jones), Dan battles Wade's henchman (Richard Jaeckel), the murder victim's revenge-minded brother, and the temptation of the large bribe Wade offers in exchange for his freedom.", 
    "thumb": "/library/metadata/xxxx/thumb/xxxxxxxx", 
    "title": "3:10 to Yuma", 
    "type": "movie", 
    "updatedAt": 1614242261, 
    "viewCount": 1, 
    "year": 1957
  }, 
  "Player": {
    "local": true, 
    "publicAddress": "xx.xx.xx.xx", 
    "title": "Chrome", 
    "uuid": "xxxxxxx"
  }, 
  "Server": {
    "title": "xxxxxx", 
    "uuid": "xxxxxxx"
  }, 
  "event": "media.scrobble", 
  "owner": true, 
  "user": true
}
```

File: metadata.go [hunk 1]
```diff
@@ -13,68 +13,73 @@ import (
...
-	ViewOffset            int                `json:"viewOffset"`            // movie
-	Writer                []MetadataItem     `json:"Writer"`                // movie + show
-	Year                  int                `json:"year"`                  // movie + show
+	AddedAt               time.Time           `json:"addedAt"`               // movie + show + music
+	Art                   string              `json:"art"`                   // movie + show
+	AttributionLogo       *url.URL            `json:"attributionLogo"`       // streaming movie
+	AudienceRating        float64             `json:"audienceRating"`        // movie
+	AudienceRatingImage   string              `json:"audienceRatingImage"`   // movie
+	Banner                *url.URL            `json:"banner"`                // movie
+	Channel               []MetadataItem      `json:"Channel"`               // tv movie
```
Comment (russorat) [inline metadata.go:22]: new

File: metadata.go [hunk 2]
```diff
@@ -13,68 +13,73 @@ import (
...
+	Key                   string              `json:"key"`                   // movie + show + music
+	LastRatedAt           time.Time           `json:"lastRatedAt"`           // movie + show + music
+	LastViewedAt          time.Time           `json:"lastViewedAt"`          // movie + show + music
+	LibrarySectionID      int                 `json:"librarySectionID"`      // movie + show + music
+	LibrarySectionKey     string              `json:"librarySectionKey"`     // movie + show + music
+	LibrarySectionTitle   string              `json:"librarySectionTitle"`   // movie + show + music
+	LibrarySectionType    LibrarySection      `json:"librarySectionType"`    // movie + show + music
+	Live                  string              `json:"live"`                  // show
+	Mood                  []MetadataItem      `json:"Mood"`                  // music
+	OneShot               string              `json:"oneShot"`               // tv movie
```
Comment (russorat) [inline metadata.go:52]: new (i have no idea what this is)

File: metadata.go [hunk 3]
```diff
@@ -13,68 +13,73 @@ import (
...
+	Channel               []MetadataItem      `json:"Channel"`               // tv movie
+	ChapterSource         string              `json:"chapterSource"`         // show (movie too ?)
+	Collection            []MetadataItem      `json:"Collection"`            // movie
+	ContentRating         string              `json:"contentRating"`         // movie + show
+	Country               []MetadataItem      `json:"Country"`               // movie
+	CreatedAtAccuracy     string              `json:"createdAtAccuracy"`     // photo
+	CreatedAtTZOffset     string              `json:"createdAtTZOffset"`     // photo
+	Director              []MetadataItem      `json:"Director"`              // movie + show
+	Duration              time.Duration       `json:"duration"`              // movie
+	Field                 []MetadataItemField `json:"Field"`                 // tv movie
```
Comment (russorat) [inline metadata.go:31]: new (no idea what this is)

File: metadata.go [hunk 4]
```diff
@@ -13,68 +13,73 @@ import (
...
+	AudienceRating        float64             `json:"audienceRating"`        // movie
+	AudienceRatingImage   string              `json:"audienceRatingImage"`   // movie
+	Banner                *url.URL            `json:"banner"`                // movie
+	Channel               []MetadataItem      `json:"Channel"`               // tv movie
+	ChapterSource         string              `json:"chapterSource"`         // show (movie too ?)
+	Collection            []MetadataItem      `json:"Collection"`            // movie
+	ContentRating         string              `json:"contentRating"`         // movie + show
+	Country               []MetadataItem      `json:"Country"`               // movie
+	CreatedAtAccuracy     string              `json:"createdAtAccuracy"`     // photo
+	CreatedAtTZOffset     string              `json:"createdAtTZOffset"`     // photo
```
Comment (russorat) [inline metadata.go:28]: new for photo metadata

**Comment by russorat** [comment]
ok, i think i've found as many as i can for now! in case you were curious, i'm using this in https://github.com/russorat/telegraf-webhooks-plex

**Comment by hekmon** [comment]
> ok, i think i've found as many as i can for now!

Great ! I am not using these content types, I would never have stumble upon them :/

> in case you were curious, i'm using this in https://github.com/russorat/telegraf-webhooks-plex

Nice !

**Comment by hekmon** [comment]
Thanks again @russorat ! I am holding just a bit more before tagging v1.0.0 just in case you find something else. Don't hesitate to tell me if/when you think it is ok.

**Pull request merged**